ffi/cffi-test.c

```diff
@@ -45,36 +36,64 @@ int main(int argc, char* argv[]) {
   uint64_t v = version(snapshot);
   printf("version: %" PRIu64 "\n", v);
   ExternResultHandleSharedScan scan_res = scan(snapshot, engine, NULL);
-  if (scan_res.tag != OkHandleSharedScan) {
+  if (scan_res.tag != OkHandleSharedScan)
+  {
     printf("Failed to create scan\n");
     return -1;
   }
 
   SharedScan *scan = scan_res.ok;
 
   ExternResultHandleSharedScanDataIterator data_iter_res =
-    kernel_scan_data_init(engine, scan);
-  if (data_iter_res.tag != OkHandleSharedScanDataIterator) {
+      kernel_scan_data_init(engine, scan);
+  if (data_iter_res.tag != OkHandleSharedScanDataIterator)
+  {
     printf("Failed to construct scan data iterator\n");
     return -1;
   }
 
   SharedScanDataIterator *data_iter = data_iter_res.ok;
 
   // iterate scan files
-  for (;;) {
+  for (;;)
+  {
     ExternResultbool ok_res = kernel_scan_data_next(data_iter, NULL, visit_data);
-    if (ok_res.tag != Okbool) {
+    if (ok_res.tag != Okbool)
+    {
       printf("Failed to iterate scan data\n");
       return -1;
-    } else if (!ok_res.ok) {
+    }
+    else if (!ok_res.ok)
+    {
       break;
     }
   }
 
   drop_scan(scan);
   drop_snapshot(snapshot);
   drop_engine(engine);
+}
+
+int main(int argc, char *argv[])
+{
+
+  if (argc < 2)
+  {
+    printf("Usage: %s table/path\n", argv[0]);
+    return -1;
+  }
+
+  char *table_path = argv[1];
+  printf("Reading table at %s\n", table_path);
+
+  KernelStringSlice table_path_slice = {table_path, strlen(table_path)};
+
+  ExternResultHandleSharedExternEngine default_engine_res = get_default_engine(table_path_slice, NULL);
```

Comment:
Aside: This is (already existing) undefined behavior!

`get_default_engine` takes a function pointer as its second arg, and function pointers in rust are non-nullable. Most likely, the code would seg fault if we ever actually triggered an error, but technically UB means the bad thing that happens is up to whatever random bad choices the compiler makes because we broke invariants it relies on.

If we really want to allow this, we need to make that arg be `Option<AllocateErrorFn>` instead, and change the existing `impl AllocateError for AllocateErrorFn` to:
```rust
impl AllocateError for Option<AllocateErrorFn> {
    unsafe fn allocate_error(
        &self,
        etype: KernelError,
        msg: KernelStringSlice,
    ) -> *mut EngineError {
        if let Some(f) = self {
            f(etype, msg)
        } else {
            let msg = unsafe { String::try_from_slice(msg) };
            panic!("{etype:?}: {msg}");
        }
    }
}
```